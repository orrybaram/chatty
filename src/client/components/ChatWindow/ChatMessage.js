import React, { Component, PropTypes } from 'react';
import { observer } from 'mobx-react';
import moment from 'moment';

const propTypes = {
  message: PropTypes.object
};

@observer
class ChatMessage extends Component {
  constructor() {
    super();
    this.interval = null;
  }
  componentWillUnmount() {
    clearInterval(this.interval);
    this.interval = null;
  }
  componentDidMount() {
    this.interval = setInterval(() => {
      this.setState({});
    }, 1000);
  }
  render() {
    const { message } = this.props;
    return (
      <div className='chat-window__messages__message'>
        {message.body}
        {message.sentBy.name}
        {moment(message.dateCreated).fromNow()}
      </div>
    );
  }
}

ChatMessage.propTypes = propTypes;
export default ChatMessage;
