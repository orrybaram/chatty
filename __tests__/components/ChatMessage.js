/* eslint-disable import/no-extraneous-dependencies */
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import ChatMessage from '../../src/client/components/ChatWindow/ChatMessage';

describe('ChatMessage', () => {
  it('renders correctly', () => {
    const message = {
      body: 'yo',
      dateCreated: new Date(),
      sentBy: { name: 'gary' }
    };
    const component = renderer.create(<ChatMessage message={message} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('refreshes every X amount of miliseconds', () => {
    const message = {
      body: 'yo',
      dateCreated: new Date(),
      sentBy: { name: 'bags' }
    };

    const chatMessage = mount(<ChatMessage message={message} refreshRate={100} />);
    expect(chatMessage.state('renderCount')).toBe(1);
    setTimeout(() => {
      expect(chatMessage.state('renderCount')).toBe(2);
    }, 200);

    const instance = chatMessage.instance();
    expect(instance.interval).toBeTruthy();
    chatMessage.unmount();
    expect(instance.interval).toBe(null);
  });
});
