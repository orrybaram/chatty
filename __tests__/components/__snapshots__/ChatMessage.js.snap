exports[`ChatMessage renders correctly 1`] = `
<div
  className="chat-window__messages__message">
  <div
    className="chat-window__messages__message-wrapper">
    <div
      className="chat-window__messages__message__body">
      yo
    </div>
    <div
      className="chat-window__messages__message__date">
      a few seconds ago
    </div>
  </div>
</div>
`;
